### PR TITLE
Make path times public.

### PR DESCRIPTION
**Pull request by jess-moss**
Fixes #

Changes in this PRs:
- 
- 

Checklists:
- [x] Update HISTORY.md with a single line describing this PR

---
This change is [Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197)

### CODE REVIEW

File: cpp/src/toppra/parametrizer/const_accel.hpp [hunk 1]
```diff
@@ -16,6 +16,8 @@ class ConstAccel : public Parametrizer {
  public:
   ConstAccel(GeometricPathPtr path, const Vector &gridpoints, const Vector &vsquared);
 
+  const Vector get_m_ts() {return m_ts;}
```
Comment (jmirabel) [inline cpp/src/toppra/parametrizer/const_accel.hpp]:
```suggestion
  const Vector& getTs() { return m_ts; }
```

**Review by jess-moss — COMMENTED** [review]
Per Issue #187 

> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197#-:-MxoJUbD4JuO1g66yYJT:b7hkodg)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @jess-moss and @jmirabel)

**Review by jess-moss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @jmirabel)

___
*[cpp/src/toppra/parametrizer/const_accel.hpp, line 19 at r1](https://reviewable.io/reviews/hungpham2511/toppra/197#-MxoIrIMNQ8DmDEomVLh-r1-19:-MxoJaSH40nT1GQ7Mj5u:b-896fix) ([raw file](https://github.com/hungpham2511/toppra/blob/39b321e380c2db393c25883b3b1a6581f14a914e/cpp/src/toppra/parametrizer/const_accel.hpp#L19)):*
<details><summary><i>Previously, jmirabel (Joseph Mirabel) wrote…</i></summary><blockquote>

```suggestion
  const Vector& getTs() { return m_ts; }
```
</blockquote></details>

Done.

**Comment by jmirabel** [comment]
> Per Issue #187

Thanks for pointing the issue. It made me realize I overlooked the changes. The prototype should be `const Vector getTimes() const` so two additional changes:
- **const** to help the compiler (although it should be able to understand it)
- function name change to try to stick to the naming convention followed elsewhere in this lib (and `Ts` is a little obscure).

**Review by jess-moss — COMMENTED** [review]
Thank you for reviewing this! I need this function to gain access to these times while using toppra, so this will be very helpful.

> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197#-:-MxoNLwO7h1b2r7su2Hl:b-m756bt)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @jmirabel)

File: cpp/src/toppra/parametrizer.hpp [hunk 2]
```diff
@@ -52,6 +52,8 @@ class Parametrizer : public GeometricPath {
 
   virtual ~Parametrizer() {}
 
+  virtual const Vector& getTimes() const = 0;
```
Comment (jmirabel) [inline cpp/src/toppra/parametrizer.hpp:57]:
```suggestion
    /** \brief Return the waypoints time.
   */
   virtual const Vector& getTimes() const = 0;
```

File: cpp/src/toppra/parametrizer/const_accel.hpp [hunk 3]
```diff
@@ -16,6 +16,8 @@ class ConstAccel : public Parametrizer {
  public:
   ConstAccel(GeometricPathPtr path, const Vector &gridpoints, const Vector &vsquared);
 
+  const Vector& getTimes() const final { return m_ts; }
```
Comment (jmirabel) [inline cpp/src/toppra/parametrizer/const_accel.hpp]:
```suggestion
  const Vector& getTimes() const override { return m_ts; }
```
Why `final` ? I think `override` is sufficient here. It is fine if a child class overrides this.

Comment (jmirabel) [inline cpp/src/toppra/parametrizer/const_accel.hpp]:
```suggestion
  const Vector& getTimes() const override { return m_ts; }
```
Why `final` ? I think `override` is sufficient here. It is fine if a child class overrides this.

**Review by jmirabel — COMMENTED** [review]
Thanks. I have a few more comments.

**Review by jess-moss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @jmirabel)

___
*[cpp/src/toppra/parametrizer.hpp, line 55 at r3](https://reviewable.io/reviews/hungpham2511/toppra/197#-MxoQtNAwTenJBtJjZc9-r3-55:-MxtPO4z8IjPEfMtfgff:b-896fix) ([raw file](https://github.com/hungpham2511/toppra/blob/32769ca3732934a7fd642e5d289be155dd08b80a/cpp/src/toppra/parametrizer.hpp#L55)):*
<details><summary><i>Previously, jmirabel (Joseph Mirabel) wrote…</i></summary><blockquote>

```suggestion
    /** \brief Return the waypoints time.
   */
   virtual const Vector& getTimes() const = 0;
```
</blockquote></details>

Done.
___
*[cpp/src/toppra/parametrizer/const_accel.hpp, line 19 at r3](https://reviewable.io/reviews/hungpham2511/toppra/197#-MxoIrIMNQ8DmDEomVLh-r3-19:-MxtPOm2Aplwjsz0ZHXM:blo547d) ([raw file](https://github.com/hungpham2511/toppra/blob/32769ca3732934a7fd642e5d289be155dd08b80a/cpp/src/toppra/parametrizer/const_accel.hpp#L19)):*
<details><summary><i>Previously, jmirabel (Joseph Mirabel) wrote…</i></summary><blockquote>

```suggestion
  const Vector& getTimes() const override { return m_ts; }
```
Why `final` ? I think `override` is sufficient here. It is fine if a child class overrides this.
</blockquote></details>

I made it `final` because I could not foresee an instance when a child class would want to override this function, and didn't want a child class to do so accidentally. I can change it to `override`if you think that is fine!

**Review by jess-moss — COMMENTED** [review]
Thank you!

> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197#-:-Mxta-u06qJaEX2Bibpo:b-9755fn)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @jmirabel)

**Review by jmirabel — APPROVED** [review]

**Review by jess-moss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @jmirabel)

___
*[cpp/src/toppra/parametrizer.hpp, line 55 at r3](https://reviewable.io/reviews/hungpham2511/toppra/197#-MxoQtNAwTenJBtJjZc9-r3-55:-My9qqdX1LLDdXece8wO:bn1ytkk) ([raw file](https://github.com/hungpham2511/toppra/blob/32769ca3732934a7fd642e5d289be155dd08b80a/cpp/src/toppra/parametrizer.hpp#L55)):*
<details><summary><i>Previously, jess-moss wrote…</i></summary><blockquote>

Done.
</blockquote></details>

@jmirabel Would it be possible to resolve this comment?

**Comment by hungpham2511** [comment]
Hi @jess-moss, thanks for the PR! Could you also add a line to the [changelog](https://github.com/jess-moss/toppra/blob/make_path_times_public/HISTORY.md)?

Thanks @jmirabel for the review. The PR looks good to me; I will be happy to merge once all comments are resolved.

**Review by jess-moss — COMMENTED** [review]
Just updated the changelog! Thank you!

> *[Reviewable](https://reviewable.io/reviews/hungpham2511/toppra/197#-:-MyD4usI85DSyImABpbR:bh2k64o)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @jmirabel)

**Pull request merged**